packages/gatsby-plugin-nginx/src/nginx-generator.ts

```diff
@@ -115,23 +111,26 @@ function convertToPath(path: string) {
   return path.replace(/:splat/g, '$1')
 }
 
-function validateRedirect({ toPath }: Redirect): boolean {
+function parseRewrite({
+  toPath,
+  statusCode = 200,
+}: Redirect): 'proxy' | 'rewrite' | 'error_page' {
   try {
     new URL(toPath)
+
+    return 'proxy'
   } catch (ex) {
-    throw new Error(`redirect toPath "${toPath}" must be a valid absolute URL`)
+    // This wasn't a proxy url. It must be an internal redirect
   }
 
-  return true
-}
+  try {
+    // Parse as an internal redirect
+    new URL(toPath, 'http://example.org')
```

Comment:
why do u are using example.org?

```suggestion
    new URL(toPath)
```

Reply:
because in here, the `toPath` variable is something like `/foo/bar` . If I do something like `new URL('/foo/bar')` it will throw. 
The intent of this code is to know if the toPath is a valid `path`. To verify if a path is valid, I need to add a fake host.